draft-ietf-jsonpath-base.md

```diff
@@ -1203,11 +1203,17 @@ A relative path, beginning with `@`, refers to the current array element or memb
 filter selector iterates over the array or object.
 
 A singular path by itself in a Boolean context is an existence test which yields true if the path selects a node and yields false if the path does not select a node.
-To be more specific about the actual value of a node selected by a path, explicit comparisons are necessary. This existence test — as an exception to the general rule — also works with nodes with structured values.
+This existence test — as an exception to the general rule — also works with nodes with structured values.
+To test the value of a node selected by a path, an explicit comparison is necessary.
+For example, to test whether the node selected by the path `@.foo` has the value `null`, use `@.foo == null` (see {{null-semantics}})
+rather than the non-existence test `!@.foo` (which yields false if `@.foo` selects a node, regardless of the node's value).
 
 When a path resulting in an empty nodelist appears on either side of a comparison, the comparison yields
-true if and only if the comparison operator is `==`, `>=` or `<=` and the other side of the comparison is also a path
-resulting in an empty nodelist.
+true if and only if:
+
+* the comparison operator is `==`, `>=` or `<=` and the other side of the comparison is also a path
+resulting in an empty nodelist, or
+* the comparison operator is `!=` and the other side of the comparison is not also a path resulting in an empty nodelist.
```

Comment:
I'd rather have `!=` `<` `>` be defined as the opposite of `==` `>=` `<=`.

Reply:
Yes, I have a lot of sympathy for that. Before I make the change, let's see if there is a consensus for https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/212#issuecomment-1186563259 as this may be relevant here.

Reply:
Consider a path `p` which results in an empty nodelist. Then according to the spec, `p <= 1` and `p >= 1` yield false.

If we define `>`, `<` as the negation of the operators `<=`, `>=`, respectively, then `p > 1` and `p < 1` yield true. This breaks the normal partial order laws (`p < 1` implies `1 > p`, but then we have `1 > p` and `p > 1`, which breaks the asymmetry law that if `a < b` then not `b < a`).

Instead, I'm thinking of keeping any path `p` which results in an empty nodelist **outside** the partial order. Thus `p <= 1`, `p >= 1`, `p > 1`, and `p < 1` would all yield false (as would `p < p` and `p > p`, as expected).